docs/site/Parsing-requests.md

```diff
@@ -0,0 +1,159 @@
+---
+lang: en
+title: 'Parsing requests'
+keywords: LoopBack 4.0, LoopBack 4
+tags:
+sidebar: lb4_sidebar
+permalink: /doc/en/lb4/Parsing-requests.html
+summary:
+---
+
+## Parsing Requests
+
+This is an action in the default HTTP sequence, it parses arguments from an
+incoming request and uses them as inputs to invoke the corresponding controller
+method.
+
+This action contains 3 steps:
+
+- Parses arguments from request query, body, path and header according to the
+  operation's OpenAPI specification.
+- Coerces parameters from string to its corresponding JavaScript run-time type.
+- Performs validation on the parameters and body data.
+
+### Parsing Raw Data
+
+The code below defines a typical endpoint by decorating a controller method with
+rest decorators.
+
+```ts
+class TodoController() {
+  @put('/todos/{id}')
+  async replaceTodo(
+    @param.path.number('id') id: number,
+    @requestBody() todo: Todo,
+  ): Promise<boolean> {
+    return await this.todoRepo.replaceById(id, todo);
+  }
+}
+```
+
+An operation specification will be generated in-memory to describe it, and raw
+datas are parsed from request according to the specification. For example the
+first parameter is from source `path`, then its value will be parsed from a
+request's path.
+
+_See [controller](Controller.md) for more details of defining an endpoint._
+
+_See
+[OpenAPI operation object](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.1.md#operationObject)
+to know more about its structure._
+
+### Coercion
+
+The parameters parsed from path, header, query of a http request are always in
+the string format. Therefore when invoking a controller function, a parameter
+need to be converted to its corresponding JavaScript runtime type, which is
+inferred from its parameter specification.
+
+Give the example operation `replaceTodo` in section
+[parsing raw data](#parsing-raw-data), which takes in a number `id` as the first
+input, coercion means users don't need to do a type conversion by themselves in
```

Comment:
period instead of comma here.
I also think the latter sentence can be better phrased. Maybe something like `Without coercion, `id` would have to be manually cast into the number type before it can be used as seen below:`